### ISSUE TITLE
issue with not starting up the server probaly after update/restart

### ISSUE THREAD

**Issue by DirtyNightmare** [prom3theu5/ConanExilesServerUpdater#12]
i have an issue with is not starting up the server probaly from the serverupdater but it works probaly when i doing it from my batch as `start ConanSandboxServer.exe -log -QueryPort=27016`

not sure if it is caused by problems in the startup param as the queryports is diffrent in the logs
`LogInit: Command line: -log -QueryPort=27016`

```
{
  "Twitch": {
    "OAuth": "",
    "Username": "",
    "Channel": "0"
  },
  "Discord": {
    "DiscordToken": "",
    "ChannelId": 0
  },
  "Conan": {
    "FolderPath": "D:\\SteamLibrary\\Servers\\conan\\",
    "Executable": "ConanSandboxServer.exe",
    "StartupParameters": "ConanSandBox?GameServerQueryPort=27016? -nosteamclient -game -server",
    "RaidingProtectionHoursEnabled": false,
    "RaidingStartHour": 0,
    "RaidingLengthInHours": 0,
    "AvatarsEnabledCertainHours": false,
    "AvatarActivationHour": 0,
    "AvatarsActiveLengthInHours": 0
  },
  "Update": {
    "ShouldInstallSteamCmdIfMissing": false,
    "ShouldInstallConanServerIfMissing": false,
    "SteamCmdPath": "D:\\SteamLibrary\\SteamCMD\\",
    "AnnounceDiscord": false,
    "AnnounceTwitch": false,
    "AnnounceMinutesBefore": 5,
    "UpdateCheckInterval": 5,
    "InstalledBuild": 1623976,
    "UpdateOnLaunch": true
  },
  "General": {
    "ShouldRestartConanOnNotRunning": true,
    "RestartServerAfterHours": 0
  }
}
```
`LogInit: Command line: ConanSandBox?GameServerQueryPort=27016? -nosteamclient -game -server -log`

**Comment by prom3theu5** [comment]
Can I see your exact batch file please, as the command line is malformed in both (you can take out any passwords)

**Comment by DirtyNightmare** [comment]
thats the `start ConanSandboxServer.exe -log -QueryPort=27016`

could it might be caused by the `?` in the config

**Comment by prom3theu5** [comment]
possibly.
Queryport isn't set like that though, so that wouldn't do anything in your batch, you can leave it out - it's the default query port anyway :)

If that's all that is in your batch - Remove everything from StartupParameters, and your server should start

**Comment by prom3theu5** [comment]
So just "StartupParameters": "",

**Comment by DirtyNightmare** [comment]
still no changes isn't the default query port 27015?

**Comment by prom3theu5** [comment]
GameServerPort = 27015
GameServerQueryPort = 27016

**Comment by prom3theu5** [comment]
change it to `"StartupParameters":"ConanSandbox?Multihome=X.X.X.X?GameServerPort=27015?GameServerQueryPort=27016?MaxPlayers=50?listen?AdminPassword=YOURADMINPASSWORD -nosteamclient -game -server"`

Make sure to change X.X.X.X to your IP, YOURADMINPASSWORD to the password you want, and you can change MAxPlayers from 50 to whatever you think you can handle up to 70

**Comment by DirtyNightmare** [comment]
multihome should be the wan ip right?

**Comment by prom3theu5** [comment]
If you are hosting for external, then yeah, most probably, int he majority of cases.
It is which ever interface you want to bind the server to.

**Comment by DirtyNightmare** [comment]
that causing it to crash
`"StartupParameters":"ConanSandbox?Multihome=mywanip?GameServerPort=27015?GameServerQueryPort=27016?MaxPlayers=40?listen?AdminPassword=mypasword -nosteamclient -game -server"`

**Comment by DirtyNightmare** [comment]
that are giving me the 
```
[2017.02.11-03.18.31:050][  1]LogOnline:Error: Failed to bind listen socket to addr (85.191.250.119:14001) for LAN beacon
[2017.02.11-03.18.31:050][  1]LogOnline:Warning: Failed to init to LAN beacon 
```

`"StartupParameters":"ConanSandbox?Multihome=85.191.250.119?GameServerPort=27015?GameServerQueryPort=27016?MaxPlayers=40?listen?AdminPassword=mypassword-nosteamclient -game -server",`

i saw somewhere on steam that people said that multihome doesn't work if you wanna have it public

**Comment by prom3theu5** [comment]
I use it on mine, but my server has 32 External IPs :)
You can leave it out and it'll bind it automatically

`"StartupParameters":"ConanSandbox?GameServerPort=27015?GameServerQueryPort=27016?MaxPlayers=40?listen?AdminPassword=mypassword-nosteamclient -game -server",`

**Comment by prom3theu5** [comment]
you need a space after your admin password. before -nosteamclient too btw

**Comment by DirtyNightmare** [comment]
strange it will only come up when i launch it from the batch 
i also wonder if you could might add a setting to close steam client before starting up the server and start steam again as that causing some issues with missing dll

**Comment by Aoxmodeus** [comment]
Ok, checkit out.  Multihome tells the app what LOCAL IP to bind to.  This must be an ip address on the machine.  You'll then use port forwarding of some sort on your router to forward your external IP to the internal.  If you try to bind to an external live IP hosted on your router, that breaks the stuff.  It must bind to an address on one of your NICs.

**In startup parameters, get rid of "ConanSandbox?" and just start it with your parameters.**  If you've defined these parameters in the saved config on the server, you do not need to specify them in the config for startup.    The config files in question are in: \exiles\ConanSandbox\Saved\Config\WindowsServer, and specifically the ones that hold settings set by the startup paramters are in Game.ini, Engine.ini and ServerSettings.ini. 

I've never used -game or -server on my successful implementations, so you can probably nix those, and default port is 27015, so you can probably nix that as well.

**Comment by Aoxmodeus** [comment]
For instance, my -parameters consist soley of "-MULTIHOME:192.168.50.32 -nosteamclient".  I have multiple ips on the server hosting the conan process.  It has a "base" address configured in the windows TCP/IP config fields, but I've added ips to it for multiple purposes.  Since the conan process doesn't know which IP address to use, it defaults to using the first configured address. If you want to use a different IP that you've added to the box, you need to tell the conanserver to use that new IP address.  That's what the multihoming command does.

If your system only has one IP, then you don't need this command.  You only need to use multihome on a "multhomed" system, one that has multiple IP addresses.

**TL:DR Get rid of Conansandbox? and just use "-log -nosteamclient" for your paramters if you only have one IP, and see if that works.**

**Comment by DirtyNightmare** [comment]
theese params did the trick 
`"StartupParameters":"-QueryPort=27016 -log -nosteamclient",`
the only way it could get up was only by using `-QueryPort=27016`

**Comment by prom3theu5** [comment]
Glad you got it sorted. Thanks Aox, I had to sleep. EU, so was late :)

**Comment by Aoxmodeus** [comment]
No worries buddy, I'm in California and a night owl, easy to jump in and help when I can.
